### ISSUE TITLE
UMat support for cv::bilateralFilter with float32

### ISSUE THREAD

**Issue by carter54** [opencv/opencv#20853]
##### System information (version)
OpenCV => 4.5.3
Operating System / Platform => Windows 64 Bit
Compiler => Visual Studio 2019

##### Detailed description
As discuss here
https://forum.opencv.org/t/umat-support-float32/5428

UMat support for cv::bilateralFilter with float32
cv::bilateralFilter does not have an OpenCL path for float 32 data type (only support int 8).
https://github.com/opencv/opencv/blob/39c3334147ec02761b117f180c9c4518be18d1fa/modules/imgproc/src/bilateral_filter.dispatch.cpp#L403

is it possible to add support for float 32?

---

relates #20760

**Comment by kriswgc** [comment]
No, cv::bilateralFilter with float 32 of opencl function is not supported .
Only u8 type while opencl :bilateralFilter, if the dept of UMat is not u8, the cv::bilateralFilter will use generic opencv api , not opecl api.
